### ISSUE TITLE
Make a pass over SITL sensor pipeline and tune models

### ISSUE THREAD

**Issue by RomanBapst** [PX4/PX4-Autopilot#10529]
- [ ] delta_wing
 - [ ] iris
 - [ ] iris_fpv_cam
 - [ ] iris_irlock
 - [ ] iris_opt_flow
 - [ ] iris_rplidar
 - [ ] iris_vision
 - [ ] matrice_100
 - [ ] matrice_100_opt_flow
 - [ ] plane
 - [ ] polaris_ranger_ev
 - [ ] rover
 - [ ] solo
 - [ ] standard_vtol
 - [ ] tailsitter
 - [ ] tiltrotor
 - [ ] typhoon_h480

**Comment by dagar** [comment]
https://github.com/PX4/Firmware/commits/pr-jenkins_sitl_models

 - typhoon_h480 working
 - delta_wing badly broken
 - iris_opt_flow needs more thought

![image](https://user-images.githubusercontent.com/84712/45772095-dabd3180-bc14-11e8-9e8f-c4a23fd0da67.png)

**Comment by RomanBapst** [comment]
Some issues I noticed:
- tailsitter in the CI mission does the transition and goes back to the previous waypoint before navigating to the landing waypoint -> tests times out because it takes too long
- innovation check often fail for fw tests, because of the impact resulting from ground contact during landing
- generally innovations become quite large during transitions -> during a transition there are high accelerations and it could be an issue with wrong measurement delay for the ekf or inconsistent sensor data (wrong dt, ...)

**Comment by RomanBapst** [comment]
@dagar So one sure cause for the failures we see is the fact that the position controller (ever since flight task was merged) is totally unaware of any VTOL related stuff. That means that setpoints are not reset properly after a transition. We need to discuss how a flighttask should support transitions.
The tailsitter failure which happens regularly is actually because after the backtransition the vehicle tries to catch up with the position setpoint. Doing that it loses time and as a result is does not finish the mission in time.

**Comment by RomanBapst** [comment]
@dagar This is another interesting one. It seems like the vehicle is never able to reach the 2. last waypoint and as a result it keeps circling around it forever.
This could be a real issue. Generally the mission in the CI is very small.
![image](https://user-images.githubusercontent.com/7610489/45874859-4084f700-bd96-11e8-82ab-a4a2ff51415b.png)

**Comment by julianoes** [comment]
I can't get `standard_vtol` to fly properly even headless on macOS.
It looses the position estimate, climbs like crazy and then does a failsafe land.

```
INFO  [logger] Start file log
INFO  [logger] Opened log file: rootfs/fs/microsd/log/2018-11-07/08_23_07.ulg
INFO  [ecl/EKF] EKF aligned, (pressure height, IMU buf: 22, OBS buf: 14)
INFO  [mavlink] partner IP: 127.0.0.1
INFO  [ecl/EKF] EKF GPS checks passed (WGS-84 origin set)
INFO  [ecl/EKF] EKF commencing GPS fusion
INFO  [mavlink] partner IP: 127.0.0.1
[Wrn] [Publisher.cc:141] Queue limit reached for topic /gazebo/default/standard_vtol/gazebo/command/motor_speed, deleting message. This warning is printed only once.
INFO  [commander] Takeoff detected
WARN  [commander_tests] Failsafe enabled: no global position
WARN  [navigator] Global position failure: fixed bank loiter
WARN  [ecl/EKF] EKF GPS fusion timeout - reset to GPS
[Wrn] [Publisher.cc:141] Queue limit reached for topic /gazebo/default/standard_vtol/groundtruth, deleting message. This warning is printed only once.
WARN  [ecl/EKF] EKF GPS fusion timeout - reset to GPS
WARN  [commander] Landing denied, land manually
WARN  [ecl/EKF] EKF GPS fusion timeout - reset to GPS
```

<img width="1099" alt="screen shot 2018-11-07 at 09 27 42" src="https://user-images.githubusercontent.com/1419688/48119498-b8887d00-e26f-11e8-82f4-68fbc5274990.png">

**Comment by RomanBapst** [comment]
@julianoes Let me try to replicate it quickly, I flew a few days ago and it was still fine.

**Comment by RomanBapst** [comment]
@julianoes No issues on Ubuntu. Can you describe what exactly you do, mission?

**Comment by hamishwillee** [comment]
@RomanBapst @julianoes did specifically mention macOS. FWIW I tried this yesterday on Ubuntu 16.04 in headless mode on a VM and it [worked great](https://sdk.dronecode.org/en/examples/transition_vtol_fixed_wing.html). But

**Comment by julianoes** [comment]
@hamishwillee @RomanBapst yes you're right it's a macOS specific issue.

**Comment by Seanmatthews** [comment]
Not sure where this task is at, but I've had problems with `libgazebo_opticalflow_plugin.so` while using gazebo9. Trying to spawn the iris_opt_flow or iris_rplidar gets me a segfault with backtrace:

> #0  0x0000000000000000 in ?? ()
> #1  0x00007fffd1cc89b8 in cv::Mat::copyTo(cv::_OutputArray const&) const () from /usr/lib/x86_64-linux-gnu/libopencv_core.so.2.4
> #2  0x00007ffef1df5c75 in trackFeatures(cv::Mat const&, cv::Mat const&, std::vector<cv::Point_<float>, std::allocator<cv::Point_<float> > >&, std::vector<cv::Point_<float>, std::allocator<cv::Point_<float> > >&, std::vector<int, std::allocator<int> >&, int) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/OpticalFlow/libOpticalFlow.so
> #3  0x00007ffef1df30f5 in OpticalFlowOpenCV::calcFlow(unsigned char*, unsigned int const&, int&, float&, float&) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/OpticalFlow/libOpticalFlow.so
> #4  0x00007fff049bcc86 in gazebo::OpticalFlowPlugin::OnNewFrame(unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/libgazebo_opticalflow_plugin.so
> #5  0x00007fff049cf900 in boost::_mfi::mf5<void, gazebo::OpticalFlowPlugin, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::operator()(gazebo::OpticalFlowPlugin*, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/libgazebo_opticalflow_plugin.so
> #6  0x00007fff049ce5db in void boost::_bi::list6<boost::_bi::value<gazebo::OpticalFlowPlugin*>, boost::arg<1>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::operator()<boost::_mfi::mf5<void, gazebo::OpticalFlowPlugin, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>, boost::_bi::list5<unsigned char const* const&, unsigned int const&, unsigned int const&, unsigned int const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> >(boost::_bi::type<void>, boost::_mfi::mf5<void, gazebo::OpticalFlowPlugin, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>&, boost::_bi::list5<unsigned char const* const&, unsigned int const&, unsigned int const&, unsigned int const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>&, int) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/libgazebo_opticalflow_plugin.so
> #7  0x00007fff049cbbe4 in void boost::_bi::bind_t<void, boost::_mfi::mf5<void, gazebo::OpticalFlowPlugin, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>, boost::_bi::list6<boost::_bi::value<gazebo::OpticalFlowPlugin*>, boost::arg<1>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::operator()<unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(unsigned char const*&&, unsigned int&&, unsigned int&&, unsigned int&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/libgazebo_opticalflow_plugin.so
> #8  0x00007fff049c811c in std::_Function_handler<void (unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), boost::_bi::bind_t<void, boost::_mfi::mf5<void, gazebo::OpticalFlowPlugin, unsigned char const*, unsigned int, unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>, boost::_bi::list6<boost::_bi::value<gazebo::OpticalFlowPlugin*>, boost::arg<1>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<unsigned int>, boost::_bi::value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >::_M_invoke(std::_Any_data const&, unsigned char const*&&, unsigned int&&, unsigned int&&, unsigned int&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /home/dev/src/Firmware/build/px4_sitl_default/build_gazebo/libgazebo_opticalflow_plugin.so
> #9  0x00007fffedfd4117 in gazebo::rendering::Camera::PostRender() () from /usr/lib/x86_64-linux-gnu/libgazebo_rendering.so.9
> #10 0x00007ffff6038d50 in gazebo::sensors::CameraSensor::UpdateImpl(bool) () from /usr/lib/x86_64-linux-gnu/libgazebo_sensors.so.9
> #11 0x00007ffff6070d79 in gazebo::sensors::Sensor::Update(bool) () from /usr/lib/x86_64-linux-gnu/libgazebo_sensors.so.9
> #12 0x00007ffff60779a5 in gazebo::sensors::SensorManager::SensorContainer::Update(bool) () from /usr/lib/x86_64-linux-gnu/libgazebo_sensors.so.9
> #13 0x00007ffff6079ada in gazebo::sensors::SensorManager::Update(bool) () from /usr/lib/x86_64-linux-gnu/libgazebo_sensors.so.9
> #14 0x00007ffff7ba4157 in gazebo::Server::Run() () from /usr/lib/x86_64-linux-gnu/libgazebo.so.9
> #15 0x0000000000403032 in ?? ()
> #16 0x00007ffff62f4830 in __libc_start_main (main=0x402f30, argc=11, argv=0x7fffffffce68, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffce58)
>     at ../csu/libc-start.c:291
> #17 0x00000000004036a9 in _start ()
>

**Comment by julianoes** [comment]
@Seanmatthews can you make a separate issue in https://github.com/PX4/sitl_gazebo/ please? Thanks.

**Comment by Seanmatthews** [comment]
@julianoes https://github.com/PX4/sitl_gazebo/issues/292
